### DISCUSSION TITLE
Hi, how do I open a midi device without knowing its name? Is there any command to list available devices?

### DISCUSSION THREAD

**Discussion by ruiseixasm** [mido/mido#562] [Q&A] [answered]
I'm on Windows, and I would like to connect to the default windows synth device, however I don't know its name but I know it's the only one in the system, so, how to connect to device "0" using mido?

Thanks

**Comment by rdoursenaud** [comment, marked as answer]
```py
mido.open_output()
```
Should do exactly what you're looking for.
Otherwise, please check our documentation: https://mido.readthedocs.io/en/stable/ports/index.html#common
